doc/Handlers.md

```diff
@@ -0,0 +1,415 @@
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 200 OK
+< Content-Type: application/json
+< Content-Length: 4
+<
+* Connection #0 to host 127.0.0.1 left intact
+null
+```
+
+## `/append-signature`
+
+### Purpose
+
+Add the provided signature to the provided hash.
+
+### Expected data
+
+- An hash of type `BLAKE2B.t`
+- The corresponding [signature](../src/protocol/protocol_signature.ml#L3)
+
+### curl example
+
+```shell script
+$ curl -v -X POST http://127.0.0.1:4440/append-signature -d '{"hash":"c6c1aa9fd3dbb1eb8cfc319b4524ddfb6ff6e26edb196ce0310a56c5ac1d7934","signature":{"signature":"edsigtmjX41ajHPEnUTohQ4Jjy2d1LWPdSb4u2GrvwgLk4aoHFwMVDadw5zRepHyK1mcbUTJ6PmxKe4PaoyivJtYPJevuiL8iTq","public_key":"edpkuy1Qjwo4X7a3r1wA1HLAx9s5kgp7b9xUxfSpQQcTpFnqyzyrzH"}}' -H "Content-type: application/json"
+
+*   Trying 127.0.0.1:4440...
+* Connected to 127.0.0.1 (127.0.0.1) port 4440 (#0)
+> POST /append-signature HTTP/1.1
+> Host: 127.0.0.1:4440
+> User-Agent: curl/7.74.0
+> Accept: */*
+> Content-type: application/json
+> Content-Length: 273
+>
+* upload completely sent off: 273 out of 273 bytes
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 200 OK
+< Content-Type: application/json
+< Content-Length: 4
+<
+* Connection #0 to host 127.0.0.1 left intact
+null
+```
+
+## `/block-by-hash`
+
+### Purpose
+
+Retrieve block corresponding to the provided hash.
+
+### Expected data
+
+The hex representation of the Hash from which one we want to retrieve the block.
+
+### Returned data
+
+JSON of the corresponding [block](../src/protocol/block.mli#L2).
+
+### curl example
+
+```shell script
+$ curl -v -X POST http://127.0.0.1:4440/block-by-hash -d '{"hash":"d1f1f05746b178cbfafa88814cf518a38e26ff41d03e5e7c3a323584844961a4"}' -H "Content-type: application/json"
+
+*   Trying 127.0.0.1:4440...
+* Connected to 127.0.0.1 (127.0.0.1) port 4440 (#0)
+> POST /block-by-hash HTTP/1.1
+> Host: 127.0.0.1:4440
+> User-Agent: curl/7.74.0
+> Accept: */*
+> Content-type: application/json
+> Content-Length: 75
+>
+* upload completely sent off: 75 out of 75 bytes
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 200 OK
+< Content-Type: application/json
+< Content-Length: 586
+<
+* Connection #0 to host 127.0.0.1 left intact
+{
+  "hash": "d1f1f05746b178cbfafa88814cf518a38e26ff41d03e5e7c3a323584844961a4",
+  "payload_hash": "5dd264b5aa3252d55ab6295e181fec4654a5c220a653bcdb61c306e2b4ea6749",
+  "state_root_hash": "a44efb410e5be9deb38b3d80c3b750d89bcbf009292359878397c9a50dc680a5",
+  "withdrawal_handles_hash": "0e5751c026e543b2e8ab2eb06099daa1d1e5df47778f7787faab45cdf12fe3a8",
+  "validators_hash": "d139b7cda8577e16c79084a253d6e9ad5cfeb07e2e5ad46223fa0f87c500b9d8",
+  "previous_hash": "f4ae816df742ec2588c90b838481e57fc3ff0f21e6b55b9668f612413a7d6ec9",
+  "author": "tz1PmKMKTbSVcam69ssKUtmSpeBRkYPTBPvW",
+  "block_height": 373,
+  "operations": []
+}
+```
+
+## `/block-level`
+
+### Purpose
+
+Retrieve the current height of the chain
```

Comment:
I think using `Return the current block level (height) of the chain` is better than `retrieve`.

Reply:
Got it, but I think is more accurate to say
> The purpose of this endpoint is to retrieve the curent block level

instead of
> The purpose of this endpoint is to return the current block level

But I can get rid of the `purpose` sub-section which is quite the same that the `returned data` sub-section.